control-plane-agent/internal/agent.go

```diff
@@ -0,0 +1,155 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+package internal
+
+import (
+	"context"
+	"log"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	controlplane "control-plane-agent/api/control-plane"
+	"control-plane-agent/api/proxy"
+	"control-plane-agent/internal/event"
+	"control-plane-agent/internal/logic"
+	"control-plane-agent/internal/registry"
+)
+
+type Config struct {
+	IsProd bool
+}
+
+func RunAgent() {
+	// DEBUG
+	cfg := Config{IsProd: false}
+
+	// Setup logger
+	if cfg.IsProd {
+		logrus.SetLevel(logrus.InfoLevel)
+		logrus.SetFormatter(&logrus.JSONFormatter{TimestampFormat: time.RFC3339Nano})
+	} else {
+		logrus.SetLevel(logrus.TraceLevel)
+		logrus.SetFormatter(&logrus.TextFormatter{ForceColors: true})
+	}
+
+	// Setup logger for http standard output
+	log.SetOutput(logrus.StandardLogger().Writer())
+
+	logrus.Info("Mesh Control Plane Agent started")
+
+	controlPlaneAPI := controlplane.NewAPI(controlplane.Config{ListenPort: 8100})
+	proxyAPI := proxy.NewAPI(proxy.Config{ListenPort: 50051})
+
+	registry.MediaProxyRegistry.Init(registry.MediaProxyRegistryConfig{CancelCommandRequestFunc: proxyAPI.CancelCommandRequest})
+	registry.ConnRegistry.Init()
+	registry.MultipointGroupRegistry.Init()
+	registry.BridgeRegistry.Init()
+
+	err := logic.LogicController.Init()
+	if err != nil {
+		logrus.Errorf("Logic controller init err: %v", err)
+		return
+	}
+
+	err = event.EventProcessor.Init(event.EventProcessorConfig{EventHandler: &logic.LogicController})
+	if err != nil {
+		logrus.Errorf("Event processor init err: %v", err)
+		return
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	var wg sync.WaitGroup
+
```

Comment:
This code in some way is repeatable, maybe create unified function that automates operations

Reply:
Not sure which code do you refer to. Can you clarify which lines of code you consider repeatable?